dicomservice/ballerina/http_service_builder.bal

```diff
@@ -0,0 +1,269 @@
+                return dicomweb:createDicomwebError(message, httpStatusCode = http:STATUS_NOT_FOUND);
+            }
+
+            DicomContext? dicomContext;
+            any|error executionResult = ();
+
+            // A GET request could be a Search or a Retrieve transaction resource request
+            dicomweb:ResourceType? resourceType = getSearchResourceFromPath(path);
+            if resourceType is dicomweb:ResourceType { // Search resource
+                // Get path params from the path
+                map<string> pathParams = getResourcePathParams(resourceType, path);
+                // Process search resource
+                check self.dicomPreprocessor.processSearchResource(req, ctx, resourceType);
+                // Get DICOM context from HTTP context
+                dicomContext = getDicomContext(ctx);
+                if dicomContext != () {
+                    executionResult = executeSearchTransactionResource(resourceType, pathParams,
+                            dicomContext, dicomService, resourceMethod);
+                    // If execution is erroneous, update DICOM context accordingly
+                    if executionResult is error {
+                        dicomContext.setInErrorState(true);
+                        dicomContext.setErrorCode(getErrorCode(executionResult));
+                    }
+                } else {
+                    return createDicomContextNotFoundError();
+                }
+            } else { // Could be a retrieve resource
+                resourceType = getRetrieveResourceFromPath(path);
+                if resourceType is dicomweb:ResourceType {
+                    // TODO: Implement
+                    return createTransactionNotSupportedError("Retrieve transaction (WADO-RS)");
+                } else {
+                    return createInvalidResourceError(req.extraPathInfo);
+                }
+            }
+            return executionResult;
+        }
+
+        isolated resource function post [string... path](http:Request req, http:RequestContext ctx) returns any|error {
+            // Get DICOM service from the holder
+            Service dicomService = self.dicomServiceHolder.getDicomService();
+            // Get matching method in the DICOM service
+            handle? resourceMethod = getResourceMethod(dicomService, path, http:POST);
+
+            if resourceMethod == () { // No matching method
+                return createPathNotFoundError(req.extraPathInfo);
+            }
+
+            json|http:ClientError payload = req.getJsonPayload();
+            if payload is http:ClientError { // Invalid payload
+                return createInvalidPayloadError();
+            }
+
+            dicomweb:ResourceType? storeResourceType = getStoreResourceFromPath(path);
+            if storeResourceType == () { // Invalid DICOMweb resource
+                return createInvalidResourceError(req.extraPathInfo);
+            }
+
+            // TODO: Implement
```

Comment:
Track todos in a issue and link

Reply:
Added in https://github.com/ballerina-platform/module-ballerinax-health.dicom/commit/3807383514d097fe372c936aa46ef84a627a58d3